src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,162 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
-    @Override
-    public void add(T value) {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double INCREMENT_VALUE = 1.5;
+
+    private transient Object[] elementData = new Object[DEFAULT_CAPACITY];
+    private int size;
+
+    private Object[] grow(int minCapacity) {
+        int oldCapacity = elementData.length;
+        int newCapacity;
+        if (oldCapacity * INCREMENT_VALUE > minCapacity) {
+            newCapacity = (int) (oldCapacity * INCREMENT_VALUE);
+        } else {
+            newCapacity = (int) (minCapacity * INCREMENT_VALUE);
+        }
+        Object[] newArray = new Object[newCapacity];
+        System.arraycopy(elementData, 0, newArray, 0, elementData.length);
+        return elementData = newArray;
+    }
+
+    private Object[] grow() {
+        return grow(size + 1);
+    }
+
+    private void rangeCheckForAdd(int index) {
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index is incorrect " + index);
+        }
+    }
+
+    public Object[] toArray(List<T> list) {
+        if (list != null) {
+            Object[] array = new Object[list.size()];
+            for (int i = 0; i < array.length; i++) {
+                array[i] = list.get(i);
+            }
+            return array;
+        }
+        throw new RuntimeException("Given object cannot be converted to an array");
+    }
```

Comment:
The `toArray(List<T> list)` method is not part of the standard `List` interface and seems unnecessary. Consider removing it unless it's specifically required by your task.